sys/usb/usbus/usbus.c

```diff
@@ -0,0 +1,324 @@
+/*
+ * Copyright (C) 2018 Koen Zandberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup usb_usbus
+ * @{
+ * @file
+ * @brief   USBUS USB manager thread, handles USB interaction
+ *
+ * @author  Koen Zandberg <koen@bergzand.net>
+ * @}
+ */
+
+#include "thread.h"
+#include "byteorder.h"
+#include "usb/usbdev.h"
+#include "usb/descriptor.h"
+#include "usb/usbus.h"
+#include "usb/usbus/fmt.h"
+
+#include "usb.h"
+#include "cpu.h"
+
+#include <stdint.h>
+#include <string.h>
+#include <errno.h>
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#define _USBUS_MSG_QUEUE_SIZE    (8)
+extern const usbus_handler_driver_t _ep0_driver;
+
+/* Forward declaration of the generic USBUS event callback */
+static void _event_cb(usbdev_t *usbdev, usbdev_event_t event);
+/* Forward declaration of the endpoint USBUS event callback */
+static void _event_ep_cb(usbdev_ep_t *ep, usbdev_event_t event);
+
+static void *_usbus_thread(void *args);
+
+void usbus_init(usbus_t *usbus, usbdev_t *usbdev)
+{
+    /* TODO: Check if memset works/is necessary */
+    memset(usbus, 0, sizeof(usbus));
```

Comment:
Shouldn't this be `sizeof(*usbus)` ?

Reply:
Yes

Noticed it after I rebased and pushed everything :cry:

Fixed